x/concentrated-liquidity/spread_rewards.go

```diff
@@ -254,11 +255,40 @@ func (k Keeper) prepareClaimableSpreadRewards(ctx sdk.Context, positionId uint64
 	}
 
 	// Claim rewards, set the unclaimed rewards to zero, and update the position's accumulator value to reflect the current accumulator value.
-	spreadRewardsClaimed, forfeitedDust, err := updateAccumAndClaimRewards(spreadRewardAccumulator, positionKey, spreadRewardGrowthOutside)
+	spreadRewardsClaimedScaled, forfeitedDustScaled, err := updateAccumAndClaimRewards(spreadRewardAccumulator, positionKey, spreadRewardGrowthOutside)
 	if err != nil {
 		return nil, err
 	}
 
+	spreadFactorScalingFactor, err := k.getSpreadFactorScalingFactorForPool(ctx, position.PoolId)
+	if err != nil {
+		return nil, err
+	}
+
+	// We scale the spread factor per-unit of liquidity accumulator up to avoid truncation to zero.
+	// However, once we compute the total for the liquidity entitlement, we must scale it back down.
+	// We always truncate down in the pool's favor.
+	spreadRewardsClaimed := sdk.NewCoins()
+	forfeitedDust := sdk.DecCoins{}
+	if spreadFactorScalingFactor.Equal(sdk.OneDec()) {
```

Comment:
Can be precomputed

Reply:
https://github.com/osmosis-labs/osmosis/pull/7876/commits/09eed437ef02ce9846b4092cb42f6e22ad093107